File: algorithms/matrix-diagonal-sum/matrix-diagonal-sum.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function (mat) {
    const len = mat.length
    const lastIndex = len - 1
    let result = 0
    for (let i = 0; i < len; i++) {
        result += mat[i][i]
        if (lastIndex - i !== i) {
            result += mat[lastIndex - i][i]
        }
    }
    return result
};

File: algorithms/matrix-diagonal-sum/matrix-diagonal-sum.rs
impl Solution {
    pub fn diagonal_sum(mat: Vec<Vec<i32>>) -> i32 {
        let n = mat.len();
        let mut result: i32 = 0;
        for i in 0..n {
            result += mat[i][i];
            if (n - i - 1 == i) {
                continue;
            }
            result += mat[n - i - 1][i];
        }
        result
    }
}
